src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -89,6 +92,35 @@ void symbolValidator(ModuleSymbol module)
             }
         }
 
+        [Fact]
+        public void MultipleCallingConventions()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+#pragma warning disable CS0168
+unsafe class C
+{
+    public void M()
+    {
+        delegate* unmanaged[Thiscall, Stdcall]<void> ptr;
+    }
+}");
+
+            // PROTOTYPE(func-ptr): Should pass, verify emit and readback
+            comp.VerifyDiagnostics(
+                // (7,29): error CS8807: 'Thiscall' is not a valid calling convention specifier for a function pointer. Valid conventions are 'Cdecl', 'Stdcall', 'Thiscall', and 'Fastcall'.
+                //         delegate* unmanaged[Thiscall, Stdcall]<void> ptr;
+                Diagnostic(ErrorCode.ERR_InvalidFunctionPointerCallingConvention, "Thiscall").WithArguments("Thiscall").WithLocation(7, 29),
+                // (7,39): error CS8807: 'Stdcall' is not a valid calling convention specifier for a function pointer. Valid conventions are 'Cdecl', 'Stdcall', 'Thiscall', and 'Fastcall'.
+                //         delegate* unmanaged[Thiscall, Stdcall]<void> ptr;
+                Diagnostic(ErrorCode.ERR_InvalidFunctionPointerCallingConvention, "Stdcall").WithArguments("Stdcall").WithLocation(7, 39)
+            );
+        }
+
+        [Fact]
+        public void UnrecognizedCallingConventions()
```

Comment:
>UnrecognizedCallingConventions [](start = 20, length = 30)

It looks like this test is always going to pass.

Reply:
Whoops, meant to delete this entirely, I expanded the existing test.

---
In reply to: [452358536](https://github.com/dotnet/roslyn/pull/45684#discussion_r452358536) [](ancestors = 452358536)